packages/nextjs/hooks/scaffold-eth/useScaffoldContractRead.ts

```diff
@@ -11,7 +10,7 @@ import { BigNumber } from "ethers";
  * @param args - args to be passed to the function call
  * @param readConfig - extra wagmi configuration
  */
-export const useScaffoldContractRead = <TReturn extends BigNumber | string | boolean = any>(
+export const useScaffoldContractRead = <TReturn = any>(
```

Comment:
Not a fan of using `any`. 
We should improve the restrictions we put to allow all possible return types. This way users will better understand what it can return by looking at the allowed types.

I suggest to add a `Struct` type:

```javascript
interface Struct {
  [key: string | number]: BigNumber | string | boolean | Struct;
}
```

Then we can upate the generic type like so:

```javascript
export const useScaffoldContractRead = <TReturn extends BigNumber | string | boolean | Struct = any>
```

Then you can use the hook like so:
```javascript
const { data } = useScaffoldContractRead<{ keyName: boolean }>( ...
```

or

```javascript
const { data } = useScaffoldContractRead<[ boolean ]>( ...
```

Reply:
> Not a fan of using any.

+++ , Really love your suggestions but again user may have different return types eg. [BigNumber, string] or even [boolean] is not working and by restricting they won't be able to pass in what they think is the correct type. Also, there might be different combinations that we could miss that's why kept it loose with `any`.

I 100% agree and really don't want `any` to get into the codebase but this was a quick patch/hack until #228 is ready. My thought was if the user is passing `generic` param to `useScaffoldContractRead` then he is a bit mature and know's what's are his `return types` are.

Reply:
After exploring it a bit further and trying to expand the type to allow the various different return types (including nested structs), the type started to become so broad that it was practically `any`... So little benefit...

> My thought was if the user is passing generic param to useScaffoldContractRead then he is a bit mature and know's what's are his return types are.

Fair point :+1:

Reply:
Love this discussion. Thanks.